### ISSUE TITLE
Delay transcoding uploaded videos until after live streams have ended

### ISSUE THREAD

**Issue by CyrilBrulebois** [Chocobozzz/PeerTube#3870]
First off: thanks for introducing the live feature, it was a pleasure to experiment with it!

**Describe the current behavior**

The COAGUL association was live-streaming from OBS (using the https://peertube.stream/ instance) this Friday evening, and things were running rather smoothly, until a major issue kicked in, persisted for 1 or 2 minutes, and affected everyone. According to the instance's admin, a video had been uploaded, and it was processed right away. As far as I understand, this means encoding/transcoding it into different formats. During that time, it wasn't even possible to refresh the live's page (https://peertube.stream/videos/watch/0203bf19-7e2c-4dad-bb6d-2ebf5f7c711c) — let alone resume the stream playback…

**Steps to reproduce:**

1. Start a live stream
2. Upload another video
3. See the server so busy with processing the video that it cannot serve live streams, or even web pages

**Describe the expected behavior**

It would be perfect if processing the video would happen only after all live streams ended, or if one fears around-the-clock streams, start with a much lower priority, so that the live streams have a chance to continue uninterrupted.

**Additional information**

* PeerTube instance:
  * URL: https://peertube.stream/
  * version: 3.1.0-rc.1
  * NodeJS version: ?
  * Ffmpeg version: ?

* Browser name, version and platforms on which you could reproduce the bug: Any browsers, any platforms (all clients were affected)

**Comment by rigelk** [comment]
hi @CyrilBrulebois,

What were you live/vod settings relative to transcoding at the moment? What machine (mem/proc) were you using? Did you set VOD transcoding threads to a minimum?

> start with a much lower priority, so that the live streams have a chance to continue uninterrupted

We are already giving higher priority to live transcoding processes.

**Comment by johanfleury** [comment]
I think this relates to #947 as having transcoding done remotely would help reducing the load on the PeerTube server.

**Comment by CyrilBrulebois** [comment]
Hi @rigelk,

I have no idea about any settings server-side, and I merely have the stream key associated with the account (I would have to ask another LUG member for any settings that might be linked to our account).

I can share my client-side settings though, OBS configured with:

- Output mode: Simple (≠ Avanced)
- Video bitrate: 4000 Kbps
- Encoder: Software (x264)
- Audio bitrate: 160
- Video resolution: 1280x720
- FPS: 30

On the Peertube's stream viewer, one could select 360p or 720p.

**Comment by Chocobozzz** [comment]
Hello,

Could you forward @rigelk questions to the instance admin? And what was the other video that was uploaded at this moment?

**Comment by CyrilBrulebois** [comment]
Yes, questions have been forwarded.

**Comment by Chocobozzz** [comment]
Any news? :)

**Comment by scanlime** [comment]
For what it's worth, there's another load spike related to hashing and copying the uploaded video; even with ffmpeg reniced to a low enough priority, you may be hitting CPU or disk related bottlenecks on the main peertube nodejs process immediately after an upload and after a transcode finishes. The video ingest server is in the same nodejs process.

**Comment by CyrilBrulebois** [comment]
> Any news? :)

Just pinged the admin via a different channel (Mastodon), not sure the contact form worked last time.

**Comment by raphgilles** [comment]
Hello,

I'm the administrator of the instance.

So. I know this issue. The chances that a live and an upload are launched simultaneously are small, but sometimes it happens.

It's possibly due to how I store the videos. I'm using SSHFS with another server. A Storage Box from Hetzner.
It's not an overload issue. I think that it's a limit of a number of simultaneous connexions due to SSHFS. But I don't know how to verify ^^

**Comment by Chocobozzz** [comment]
Closing since it seems to be a system setup issue (we can't really do anything if the storage is slow)
